Data/Scripts/GardenConquest/Messaging/FleetResponse.cs

```diff
@@ -0,0 +1,48 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+using GardenConquest.Blocks;
+using GardenConquest.Extensions;
+using GardenConquest.Records;
+
+namespace GardenConquest.Messaging {
+	public class FleetResponse : BaseResponse {
+
+		public List<GridEnforcer.GridData> FleetData;
+		public GridOwner.OWNER_TYPE OwnerType;
+
+		private const int BaseSize = HeaderSize;
+
+		public FleetResponse()
+			: base(BaseResponse.TYPE.FLEET) {
+
+		}
+
+		public override byte[] serialize() {
+			VRage.ByteStream bs = new VRage.ByteStream(BaseSize, true);
+
+			byte[] bmessage = base.serialize();
+			bs.Write(bmessage, 0, bmessage.Length);
+
+			GridOwner.OWNER owner = GridOwner.ownerFromPlayerID(Destination[0]);
```

Comment:
The actual fetching of information to send in the response should be done by the method that's creating the response. The response should only deal with preparing data it's been seeded with to be sent. 

So move this creation of a GridOwner and the fetching of the corresponding fleet to line 102 (plus newline) in RequestProcessor, where they're currently being instantiated empty.